pandas/tests/indexes/multi/test_tuples.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+
+from pandas import (Index, MultiIndex)
+
+import pandas.util.testing as tm
+
+from pandas.tests.indexes.common import Base
+
+
+class TestTuples(Base):
+    _holder = MultiIndex
+    _compat_props = ['shape', 'ndim', 'size', 'itemsize']
+
+    def setup_method(self, method):
+        major_axis = Index(['foo', 'bar', 'baz', 'qux'])
+        minor_axis = Index(['one', 'two'])
+
+        major_labels = np.array([0, 0, 1, 2, 3, 3])
+        minor_labels = np.array([0, 1, 0, 1, 0, 1])
+        self.index_names = ['first', 'second']
+        self.indices = dict(index=MultiIndex(levels=[major_axis, minor_axis],
+                                             labels=[major_labels, minor_labels
+                                                     ], names=self.index_names,
+                                             verify_integrity=False))
+        self.setup_indices()
+
+    def create_index(self):
+        return self.index
+
```

Comment:
move all of the constructors (from_product, from_array, from_tuples) into a test_constructor